readthedocs/doc_builder/environments.py

```diff
@@ -348,12 +328,7 @@ def run_command_class(
         build_cmd = cls(cmd, **kwargs)
         self.commands.append(build_cmd)
         build_cmd.run()
```

Comment:
`self.pre_run_command` shuold be before this line and should receive only the `build_cmd`, `warn_only` and `**kwargs`